src/starks/constraints/evaluator.rs

```diff
@@ -85,30 +111,47 @@ impl<'poly, F: IsFFTField, A: AIR + AIR<Field = F>> ConstraintEvaluator<'poly, F
                 .unwrap()
             })
             .collect();
+        let mut boundary_evaluation =
+            vec![FieldElement::<F>::zero(); boundary_polys_evaluations[0].len()];
+
+        for col in 0..n_trace_colums {
+            let (alpha, beta) = &alpha_and_beta_boundary_coefficients[col];
+            // for each element e in columns
+            // e = e*alpha + beta
+            boundary_polys_evaluations[col] = boundary_polys_evaluations[col]
+                .iter()
+                .zip(&d_adjustment_power)
+                .map(|(v, d)| v * (alpha * d + beta))
+                .collect::<Vec<FieldElement<F>>>();
+
+            // Multiply each column for their correspondent zerofiers
+            let mut acc = boundary_polys_evaluations[col].clone();
+            // Step contains the indexes for the boundary constraints
+            // Of the respective column we are interested in
+            for column_step in boundary_constraints.steps(col).iter() {
+                let step_index = boundary_constraints
+                    .steps_for_boundary()
+                    .iter()
+                    .position(|x| *x == *column_step)
+                    .expect("Should be here");
```

Comment:
Is this logically guaranteed or could this fail on bad input? If it's the latter, we should return an error instead.